src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/SpecifyIFormatProvider.cs

```diff
@@ -309,5 +310,22 @@ static INamedTypeSymbol UnwrapNullableValueTypes(INamedTypeSymbol typeSymbol)
                 return typeSymbol;
             }
         }
+
+        private static bool IsValidParseCall(IInvocationOperation invocationOperation, INamedTypeSymbol? guidType)
+        {
+            var targetMethod = invocationOperation.TargetMethod;
+
+            if (targetMethod.Name != "Parse")
+            {
+                return false;
+            }
+
+            if (guidType != null && targetMethod.ContainingType.Equals(guidType))
+            {
+                return false;
+            }
+
+            return false;
+        }
```

Comment:
This always returns false. A test that fails before the change and passes after the change is a must in my opinion.

Normally, the testing library will need to be using Preview 4 instead of [Preview 3](https://github.com/dotnet/roslyn-sdk/blob/b7aa8cb7975555147ce03c8746fba47a00a1a0b9/src/Microsoft.CodeAnalysis.Testing/Microsoft.CodeAnalysis.Analyzer.Testing/ReferenceAssemblies.cs#L986-L991) for this to be testable.

But it might also be testable without using Preview 4 by providing empty reference assemblies and defining a fake System.Guid in the test.

Reply:
Well, now I feel foolish. 

This was the version I was testing locally to ensure that my changes did suppress the warning... and I forgot to revert it to `return true;` before pushing.

Reply:
I'm not having much success with adding a unit test. I've written this code, but the test process () appears to go into an infinite loop:

```csharp
        [Fact]
        [WorkItem(5999, "https://github.com/dotnet/roslyn-analyzers/issues/5999")]
        public async Task CA1305_GuidParse_NoDiagnosticsAsync()
        {
            await new VerifyCS.Test
            {
                ReferenceAssemblies = new ReferenceAssemblies(""), // workaround for lack of .NET 7 Preview 4 reference assemblies
                TestCode = @"
namespace System
{
    public class Object { }
    public abstract class ValueType { }
    public struct Void { }
    public class String { }
    public interface IFormatProvider { }
    public struct Guid
    {
        public static Guid Parse(string s) => default;
        public static Guid Parse(string s, IFormatProvider provider) => default;
    }
}
namespace Test
{
    using System;
    public class SomeClass
    {
        public Guid SomeMethod(string s)
        {
            return Guid.Parse(s);
        }
    }
}",
            }.RunAsync();
        }
```

Can we wait to add the test until `ReferenceAssembles.Net.Net70` are available to the test suite?

Reply:
> but the test doesn't issue a diagnostic when return true is changed back to return false (i.e., so the warning is expected).

Did you try debugging and see why this happened?

Reply:
No; I haven't been able to get the solution building successfully in VS2022, so I can't debug the tests. The README suggests VS2019, so I'll try that next. If I still can't get it working, I'll open an issue with more details.

Reply:
I used to work on the repo using VS2022, so it should work.

Reply:
Was able to get my VS2019 environment working, and trivially found the problem using the debugger. Pushed a test for the new code. @Youssef1313 thanks for the suggestion of bypassing the `ReferenceAssemblies` and manually defining a fake `System.Guid`.